### ISSUE TITLE
[BUG] Ar Tonelico 2 - Music not playing during scenes

### ISSUE THREAD

**Issue by GirlsCantKissGirls** [PCSX2/pcsx2#4504]
**Describe the bug**
During certain conversation topics between the girls at inns there is supposed to be a certain song that plays. For whatever reason the song refuses to play for me.  Somebody else has told me that the music plays for them during the scenes so I copied all of their settings identically and it still did not work.

This timestamp is what is supposed to happen.
https://youtu.be/-egoBt9RImg?t=1584

This timestamp is what happens in the emulator.
https://youtu.be/6OMXUE_9Wg4?t=166

**To Reproduce**
Just play the game normally and view the events and it will happen.

This memory card has a save for anybody to test easily. 

https://mega.nz/file/6IkRhQ7C#ZtW7ncrO-12FZmxDhX27QwKUCwLA6m3kHEx_72TSleQ

It is save #2 on the list. When you load in, talk to the shop keeper, select rest, and then select any of the girls to start a conversation, then select the topic. After that first conversation is over the two girls will start talking to each other and the music is supposed to play like the video I posted above.

**Expected behavior**
For the music to play properly.

**GS Settings**
I've tried all sorts of settings, I even set it to the preset safe default. Everything yielded the same bug.

**Emulation Settings**
Again everything was set to default to test.

**System Info (please complete the following information):**
 -   PCSX2 Revision: v1.7.0-dev-1298-gb7dfb2c29 ( I have also tried 1.4 , 1.5 , 1.6 and they all don't work )
 -   OS: Windows 10
 -   CPU: AMD Ryzen 3700x
 -   GPU: GTX 1070

**Comment by Squall-Leonhart** [comment]
did you turn on the cdvd speed hack?

**Comment by GirlsCantKissGirls** [comment]
Just tested it, it didn't do anything.

**Comment by Squall-Leonhart** [comment]
I believe your disk image is damaged.

**Comment by GirlsCantKissGirls** [comment]
I don't believe that is the case.  Have you tested it yourself to see if it works properly for you?

**Comment by arcum42** [comment]
Was able to reproduce it on my system, though I'm not discounting the possibility of my copy of the iso being damaged in exactly the same way, since while I think I grabbed my original rip, I could have grabbed a patched copy instead (since at some point, I ran an undub/translation patch on it).

I'll probably try running it directly from the original disc at some point, to see if that makes a difference, but it's too hot over by that computer right now for me to spend more time over there right now...

**Comment by Squall-Leonhart** [comment]
i've got reproductions on all 3 game regions

interpreters don't change the behavior
changing the sync type in SPU2 doesn't change it
disabled game fixes don't change anything
speed hacks on/off don't change anything.
Boot from bios (of the appropriate region) doesn't change anything.

SPU2 Logging doesn't reveal anything (probably need a debug build for more logging)

I've seen mention on the forum that character portraits and dialogs had issues in the past where some amount of corruption would present, there could be something getting skipped that cues the changing of the music track.

https://forums.pcsx2.net/Thread-Audio-problem-with-Ar-tonelico-2-SLUS-21788-New-one?highlight=ar+tonelico+2

**Comment by Squall-Leonhart** [comment]
Of course, this could be ISO related, some games are known to require a CUE file, some work better as MDF.

**Comment by arcum42** [comment]
Did a little more testing, because I thought I remembered this working properly. It works in 0.9.6 using the linuz iso cdvd plugin. It's broken in 0.9.8, either with built-in iso support or that plugin.

**Comment by Squall-Leonhart** [comment]
0.9.6 - 0.9.8 is a lot of changes, but it narrows it down to the cores

**Comment by Ziemas** [comment]
Took a quick peek at this and I don't think i suspect the SPU at least. Game seems to use ADMA for all music streaming and it's just streaming out silence during the reported scene.

**Comment by Squall-Leonhart** [comment]
The prime suspects are the VU's, and VIF recompiler between those two builds, 

MicroVU was implemented right in between them, @arcum42 does the issue replicate under 0.9.8 if you use SuperVU instead?

**Comment by refractionpcsx2** [comment]
I find it highly unlikely that the VIF or VU's (which only deal with graphics/physics) are going to be responsible for audio problems, if it is, I'll be very shocked.

**Comment by Squall-Leonhart** [comment]
> I find it highly unlikely that the VIF or VU's (which only deal with graphics/physics) are going to be responsible for audio problems, if it is, I'll be very shocked.

VU1 was used for DTS audio in NHL2002, but that might be thoroughly unique in that regards.

**Comment by refractionpcsx2** [comment]
Sounds elaborate, especially for a game as simple as Ar Tonelico, but ok.

**Comment by kozarovv** [comment]
I can't reproduce issue, music just works for me on latest dev build.

**Comment by Squall-Leonhart** [comment]
I can't reproduce it working under 0.9.6 @arcum42

**Comment by kozarovv** [comment]
That's unexpected, but.. Turn off autogamefixes, and try again.

**Comment by refractionpcsx2** [comment]
wait, a gamefix is breaking it?

**Comment by Squall-Leonhart** [comment]
> wait, a gamefix is breaking it?

Doubt it, already tried that direction, and i still can't reproduce the music transitioning under 0.9.6 XD

**Comment by kozarovv** [comment]
Yes. Nearest rounding for FPU (also positive) make music stops there.

**Comment by Squall-Leonhart** [comment]
I have gamefixes turned off, went as far as renaming the file and the emulator is using the default chop/zero, music still stops.

**Comment by Ziemas** [comment]
Disabling gamefixes makes it work for me as well.

**Comment by refractionpcsx2** [comment]
Can confirm, disabling game fixes makes it work

**Comment by arcum42** [comment]
Can confirm, turning off automatic gamefixes under either 0.9.8 or the current version fixes the issue in my tests. (Current under Linux, 0.9.6/0.9.8 under Windows.)

**Comment by Squall-Leonhart** [comment]
Disabling works for NTSC, PAL still loses music.

**Comment by refractionpcsx2** [comment]
Don't know, I don't have a legit save for PAL that I can check with, going to that Inn with any of the saves on that memcard don't seem to trigger that event

**Comment by Squall-Leonhart** [comment]
[memory8 (deathlandia tests).zip](https://github.com/PCSX2/pcsx2/files/6795806/memory8.deathlandia.tests.zip)

i replaced the GAMEFILE of the SLES save in this memcard with the one from SLUS, it loads exactly the same location as SLUS but the event loses music still.

**Comment by Squall-Leonhart** [comment]
I need to hunt down a native save at this location and test.

**Comment by refractionpcsx2** [comment]
well if you find one, please share it here.

**Comment by Squall-Leonhart** [comment]
Guess i'll be playing Ar Tonelico 2. xD

Will report back!

**Comment by refractionpcsx2** [comment]
With the PAL version, setting the EE cycle rate to +3 fixes the music. +1 or +2 breaks the game.

**Comment by arcum42** [comment]
Should be a relatively easy spot to reach in the game at least, IIRC, though it's been a while. You start with Cloche, picking up Luca is an early objective, then you have to do your first Dive with her and rest at the inn afterwards.(Though I used the original for testing, I should probably look up the latest retranslation patch and make sure my patched copy is up to date. The original translation had enough issues that fans retranslated it...)

**Comment by atomic83GitHub** [comment]
This is a very problematic edge case, the game will need softfloat or an alternative patch as nearest rounding will be needed for the treasure chests to work.

**Comment by refractionpcsx2** [comment]
Well I guess the best solution for now is to live without that particular bit of music.

**Comment by Squall-Leonhart** [comment]
the treasure chests work fine without nearest, they just temporarily drop you out of the scene if you approach them  without being dead center / hold the direction against the chest.

**Comment by refractionpcsx2** [comment]
> the treasure chests work fine without nearest, they just temporarily drop you out of the scene if you approach them without being dead center / hold the direction against the chest.

Which I personally would find more annoying than a missing bit of music :P

**Comment by atomic83GitHub** [comment]
That is still a bug in itself. Even if it minor, we can't consider it fine

**Comment by Squall-Leonhart** [comment]
so just to confirm that the PAL version of this, even with the gamefixes off cannot trigger the music change, and Refs workaround is only sporadically working, tbh the loss is acceptable as the scene it occurs on is "late at night" and any other title would play something less punchy at night time anyway.

**Comment by Mrlinkwii** [comment]
is this still an issue on latest nightly?

**Comment by Squall-Leonhart** [comment]
why would it have changed?

**Comment by JordanTheToaster** [comment]
> why would it have changed?

It's been close to 2 years and you should check if the issue is fixed.

**Comment by Squall-Leonhart** [comment]
No softfpu implementation, not fixed.

please don't make me waste time on testing based on a scenario of time passing again, do it yourself instead.

**Comment by JordanTheToaster** [comment]
> No softfpu implementation, not fixed.
> 
> please don't make me waste time on testing based on a scenario of time passing again, do it yourself instead.

This is your issue which means you need to keep track of it please keep the passive aggressive attitude to yourself thanks.

**Comment by Squall-Leonhart** [comment]
No its not, and no its not,  the onus is on the developer(s) of an application to test any changes, not on reportees 

the reporter should only be queried in the absence of a developer with the afflicted software or a missing save file.

Thanks for confirming you read none of the correspondence prior on the matter of the issue.

**Comment by stenzek** [comment]
Great, then I'm going to lock this issue. It's users like you and attitudes like this which put others off contributing, and demotivate those who are.

**Comment by refractionpcsx2** [comment]
@Squall-Leonhart talk to our testers like that again and I'll ban you from the github as well, got it?
